src/controller/agent.governance.ts

```diff
@@ -1,20 +1,28 @@
 import { Injectable } from '@nestjs/common';
 import { Logger } from 'protocol-common/logger';
+import { ProtocolException } from 'protocol-common/protocol.exception';
 import data from '../config/governence.json';
 
+// defines the callback (function) called by AgentGovernance.invokeHandler.  The signature maps to acapy
+// webhook definition
+export type HandlerCallback =
+    (agentUrl: string, agentId: string, adminApiKey: string, route: string, topic: string, body: any, token?: string) => Promise<any>;
+
+type Registration = { topic: string, func: HandlerCallback, exceptionCount: number};
+
 /**
  * TODO validation, error cases, etc
  */
 @Injectable()
 export class AgentGovernance {
-    // see GOVERANCE.md for documentation on policies data structure
     public static PERMISSION_DENY = 'deny';
     public static PERMISSION_ONCE = 'once';
     public static PERMISSION_ALWAYS = 'always';
     private static ALL_KEY = 'all';
     private static COMMENT_SECTION = 'comment';
-    private readonly policies = { };
     public policyName: string = '';
+    private readonly policies = { };
+    private readonly callbacks = new Array<Registration>();
```

Comment:
I do not see these as issues.  If a controller crashes, it will restart and startup would call the registration methods again.  We have multiple controllers now, each is its own separate code and instance.  So scaling is not an issue.

Reply:
Perhaps I misunderstood then! I thought of these callbacks as on a per-agent basis, not per-controller.